CMakeLists.txt

```diff
@@ -55,7 +56,6 @@ set(PYBIND11_HEADERS
   include/pybind11/pytypes.h
   include/pybind11/stl.h
   include/pybind11/stl_bind.h
-  include/pybind11/typeid.h
```

Comment:
That's true; it's fine with me to include it in the move.  (Maybe `type_id()` should be moved out into `cast.h` or somewhere else, but that's not really a big deal—and as an undocumented mostly internal function it was arguably misplaced by not being in `detail` in the first place).